### PR TITLE
feature: multiplexing: fix unit tests

### PR DESCRIPTION
**Pull request by fairclothjm**
This PR fixes some of the unit test failures introduced by the multiplexing feature.

The remaining test failures in `test-go-remote-docker` will be addressed in another PR.

### CODE REVIEW

File: sdk/database/dbplugin/v5/grpc_server_test.go [hunk 1]
```diff
@@ -253,13 +257,17 @@ func TestGRPCServer_NewUser(t *testing.T) {
...
+					return test.db, nil
+				},
+				instances: make(map[string]Database),
 			}
 
-			// Context doesn't need to timeout since this is just passed through
-			ctx := context.Background()
+			id := "12345"
+			idCtx := idCtx(t, id)
+			g.instances[id] = test.db
```
Comment (calvn) [inline sdk/database/dbplugin/v5/grpc_server_test.go:268]: This setup logic is being repeated more than once or twice, wondering if it'd be better to factor it out to something like:

`testGrpcServer(t *testing.T, db Database) (context.Context, gRPCServer)`

File: sdk/database/dbplugin/v5/grpc_server_test.go [hunk 2]
```diff
@@ -562,6 +586,69 @@ func TestGRPCServer_Close(t *testing.T) {
...
+	}
+
+	tests := map[string]testCase{
+		"missing plugin multiplexing metadata": {
+			ctx:          context.Background(),
+			expectedResp: "",
+			expectedErr:  fmt.Errorf("missing plugin multiplexing metadata"),
+		},
+		"unexpected number of IDs in metadata": {
+			ctx: metadata.NewIncomingContext(
```
Comment (calvn) [inline sdk/database/dbplugin/v5/grpc_server_test.go:603]: Can we use the `idCtx` helper in here?

**Review by calvn — APPROVED** [review]
Thanks for fixing the tests!

File: sdk/database/dbplugin/v5/grpc_server_test.go (hunk 1 above)
Comment (fairclothjm) [reply sdk/database/dbplugin/v5/grpc_server_test.go]: Good idea!

File: sdk/database/dbplugin/v5/grpc_server_test.go (hunk 2 above)
Comment (fairclothjm) [reply sdk/database/dbplugin/v5/grpc_server_test.go]: updated idCtx to take in variadic args

**Pull request merged**